src/auth/auth.swagger.ts

```diff
@@ -1,58 +1,165 @@
+import { applyDecorators } from '@nestjs/common';
+import { ApiOperation, ApiBody, ApiResponse, ApiTags } from '@nestjs/swagger';
 import { LoginRequestDto } from './dto/login-request.dto';
 import { SignupRequestDto } from './dto/signup-request.dto';
+import { AuthResponseDto } from './dto/auth-response.dto';
 
-export const signupSwagger = {
-  apiBody: {
-    description: '회원가입 요청 DTO',
-    type: SignupRequestDto,
-  },
-  apiCreatedResponse: {
-    description: '회원가입 성공',
-    schema: {
-      example: {
-        status: 'success',
-        message: '회원가입에 성공했습니다.',
-        data: null,
-      },
-    },
+const unauthorizedExamples = {
+  TokenExpired: {
+    message: '토큰이 만료되었습니다. 다시 로그인해주세요.',
+    error: 'Unauthorized',
+    statusCode: 401,
   },
-  apiBadRequestResponse: {
-    description: '회원가입 실패',
-    schema: {
-      example: {
-        message: '이미 사용 중인 사용자 ID입니다.',
-        error: 'Bad Request',
-        statusCode: 400,
-      },
-    },
+  InvalidToken: {
+    message: '유효하지 않은 토큰입니다.',
+    error: 'Unauthorized',
+    statusCode: 401,
+  },
+};
```

Comment:
const 어설션으로 readonly 객체로 만들어 타입스크립트 기능을 더욱 풍부하게 사용할 수 있는 방법이 있습니다

```suggestion
} as const;
```

Reply:
const 어설션을 사용해서 readonly로 만드는 방법은 처음 알았어요! 타입스크립트의 기능을 더 적극적으로 활용할 수 있겠네요. 좋은 팁 감사합니다!